ckanext/recombinant/read_xls.py

```diff
@@ -22,7 +22,73 @@ def read_xls(f, file_contents = None):
 
     row = sheet.horz_split_pos
     while row < sheet.nrows:
-        yield [c.value for c in sheet.row(row)]
+        # return next non-empty row
+        if not all(_is_bumf(c.value) for c in sheet.row(row)):
+            yield [c.value for c in sheet.row(row)]
         row += 1
 
+def _is_bumf(value):
+    """
+    Return true if this value is filler, en route to skipping over empty lines
+    """
+    if type(value) in (unicode, str):
+        return (value.strip() == '')
+    return (value is None)
+
+def _canonicalize(dirty, dstore_type):
+    """
+    Fix and return dirty input to align with recombinant.json datastore type,
+    'int' or 'text'.
+
+    The xlrd software interprets a purely numeric excel field as float,
+    and anything other as unicode, independent of recombinant.json
+    datastore type specification.
+    """
+    if dirty is None:
+        return 0.0 if dstore_type == 'int' else u''
+    elif isinstance(dirty, float) or isinstance(dirty, int):
+        if dstore_type == 'text':
+            if int(dirty) == dirty:
+                return unicode(int(dirty))
+            else:
+                return unicode(dirty)
+        return float(dirty)
+    elif (('strip' in dir(dirty)) and
+            (dirty.strip() == '') and
+            (dstore_type == 'int')):
+        return 0.0
+    elif dstore_type == 'text':
+        return unicode(dirty)
+
+    # dirty should be numeric: truncate trailing decimal digits, retain int part
+    canon = re.sub(r'[^0-9]', '', re.sub(r'\.[0-9 ]+$', '', str(dirty)))
+    try:
+        return float(canon)
+    except:
```

Comment:
can't see how this would be possible after the re.sub above (so this except block is dead code) also never use a bare except without a raise inside it (`except ValueError` would be appropriate here if it wasn't dead code)